DataStructures.Tests/BinaryTreeTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using DataStructures.Trees;
+using System.Text;
+using Xunit;
+
+namespace DataStructures.Tests
+{
+    public class TreeTests
+    {
+        [Fact]
+        public void Add_value_to_empty_tree()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(4);
+
+            //Assert
+            Assert.Equal(4, tree.Root.Value);
+        }
+
+        [Fact]
+        public void Add_one_Left_and_one_Right_node()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+
+            //Assert
+            Assert.Equal(8, tree.Root.Value);
+            Assert.Equal(4, tree.Root.Left.Value);
+            Assert.Equal(5, tree.Root.Right.Value);
```

Comment:
This assert is returning a null reference because 5 is actually placed as the right of 4.  So your assert could be "Assert.Equal(5, tree.root.Left.Right.Value);" and this test would pass.  Or see above comment about adding 10 instead of 5 and then the assert would be "Assert.Equal(10, tree.Root.Right.Value);".